Module1 complete

M-Garkaviuk opened the pull request.

M-Garkaviuk committed:
Added new app Shop, added models, configured adminsite

M-Garkaviuk committed:
Login, Logout, Registration
Admin view, product edition

M-Garkaviuk committed:
Purchase view and logic added

M-Garkaviuk committed:
Added admin processing of refunds

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -0,0 +1,150 @@
+    extra_context = {'purchase_form': PurchaseCreationForm()}
+
+
+class ProductCreateView(LoginRequiredMixin, CreateView):
```
Тоесть у тебя не админ может создать продукт?

M-Garkaviuk replied:
Закрыл через 403 создание, обновление продуктов и отображение рефаунд реквестов для не админ юзеров

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -0,0 +1,150 @@
+    success_url = '/'
+
+
+class ProductUpdate(LoginRequiredMixin, UpdateView):
```
И тут тоже

M-Garkaviuk replied:
Сделано

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -0,0 +1,150 @@
+        action = request.POST.get('action')
+
+        if action == "confirm":
+            refund.refund_purchase.product.stock += refund.refund_purchase.product_quantity
```
Тут тоже должно быть атомарное действие

M-Garkaviuk replied:
Добавил

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -0,0 +1,150 @@
+        return Purchase.objects.filter(user=user)
+
+    def get_context_data(self, **kwargs):
+        context = super().get_context_data(**kwargs)
```
А зачем тут переписан этот метод?))

M-Garkaviuk replied:
Согласен, зря. В процессе решения впихнул и забыл убрать. Я на эту страничку ничего дополлнительно не передаю. Убираю этот код

PonomaryovVladyslav reviewed (commented):
В целом ок, но я бы переписал всякий текст на месседжи, и возврат сделал атомарно. Плюс поменял пермишены все таки на админские для некоторых действий

M-Garkaviuk committed:
Removed redundant get_context_data,
added atomic transactions for refund and product update,
added success status messages,
pages for product addition, edition and refund list view restricted to staff users

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -75,17 +104,25 @@ def post(self, request, purchase_id):
     login_url = 'login/'
 
+    def dispatch(self, request, *args, **kwargs):
+        if not request.user.is_staff:
```
Тебе только тут нужна такая проверка?? И почему не сделать нормальный миксин чем вот это извращение?)

PonomaryovVladyslav commented on shop/views.py:
```diff
@@ -22,12 +25,37 @@ class ProductCreateView(LoginRequiredMixin, CreateView):
+            product = form.save(commit=False)
+            product.save()
+            messages.success(self.request, f'Product "{product.product_name}" updated successfully.')
+        return super().form_valid(form)
```
А в ретерне сейва нет?)) че ты его из атомика вытащил то?)

M-Garkaviuk committed:
-Added AdminLoginRequiredMixin to all admin related pages;
-Added indentation to ProductUpdateView;
